### PR TITLE
[SYCL] Fix the definition of const_reference in accessor

### PR DESCRIPTION
**Pull request by v-klochkov**
Signed-off-by: Vyacheslav N Klochkov <vyacheslav.n.klochkov@intel.com>

### CODE REVIEW

**Comment by v-klochkov** [comment]
It seems like it is no difference in the code, but it fixes the compilation error.
Without this 1-line fix 2 CTS tests for accessor fail with diagnostics:

./accessor_api_common_all.h:46:3: error: static_assert failed due to requirement 'std::is_same<int &, const int &>::value' "const_reference is of wrong type"
		static_assert(std::is_same<const_reference, const T &>::value

File: sycl/include/CL/sycl/accessor2.hpp [hunk 1]
```diff
@@ -333,7 +333,7 @@ class accessor :
 public:
   using value_type = DataT;
   using reference = DataT &;
-  using const_reference = const reference;
```
Comment (bader) [inline sycl/include/CL/sycl/accessor2.hpp]: Could you clarify what is wrong with the definition in the log message, please?

Comment (v-klochkov) [reply sycl/include/CL/sycl/accessor2.hpp]: The error message says that 'const_reference' is 'int &', not 'const int &' and I have no idea why.
This fix though helps to eliminate the error.

Comment (bader) [reply sycl/include/CL/sycl/accessor2.hpp]: I think it get this.
Current implementation tries to apply const qualifier to the **reference**, which doesn't make sense.
New implementation defines a reference to the `const int` type.

Comment (v-klochkov) [reply sycl/include/CL/sycl/accessor2.hpp]: I still do not understand why the previous code does not work. 'const reference' adds const qualifier to 'reference' and 'reference' is synonym to 'DataT &'. So, I would think that const add the qualifier to 'DataT &'.

In general, you agree with the fix, right?

Comment (bader) [reply sycl/include/CL/sycl/accessor2.hpp]: "<const T> &" vs "const <T&>"

> In general, you agree with the fix, right?

Yes.

**Pull request merged**